pyscada/utils/scheduler.py

```diff
@@ -730,20 +739,34 @@ def init_process(self):
         return True
 
     def loop(self):
-
+        #data from a write
+        data=[]
         # process write tasks
         for task in DeviceWriteTask.objects.filter(done=False, start__lte=time(), failed=False,
-                                                   variable__device_id=self.device_id):
+                                                   variable__device_id=self.device_id).order_by('start'):
             if task.variable.scaling is not None:
                 task.value = task.variable.scaling.scale_output_value(task.value)
-            if self.device.write_data(task.variable.id, task.value):
-                task.done = True
-                task.finished = time()
-                task.save()
+            tmp_data = self.device.write_data(task.variable.id, task.value)
+            if isinstance(tmp_data, list):
+                if len(tmp_data) > 0:
+                    task.done = True
+                    task.finished = time()
+                    task.save()
+                    data.append(tmp_data)
+                else:
+                    task.failed = True
+                    task.finished = time()
+                    task.save()
             else:
                 task.failed = True
                 task.finished = time()
                 task.save()
+        if isinstance(data, list):
```

Comment:
this will prevent the code below from execution, what was the problem you solved here?